### PR TITLE
Added Sphinx and Test Suite template

### PR DESCRIPTION
**Pull request by HagenSR**
Added templates for unit testing and sphinx documentation.

I hope that's okay!

### CODE REVIEW

**Comment by topoftheyear** [comment]
Just sphinx questions. I can't verify it myself, but have you verified it is fully functional with github pages and will slot into the main site easily, e.g. documentation pages will link to each other correctly. I only ask because the conf.py seems lighter than [prior years'](https://github.com/PixPanz/byte_le_royale_2020/blob/master/sphinx/conf.py).

I also noticed the test page files are in .rst rather than the .md used in the past. Is there a reason why? Not being accusatory, just curious.

Only changes I'd make would be to make sure attribution is made to the NDSU ACM specifically rather than just the ACM.

**Comment by HagenSR** [comment]
No problems. It does look like you can deploy sphinx to github pages (https://www.docslikecode.com/articles/github-pages-python-sphinx/). I tried doing it but the src's for css and js appear to be off so it will require some finicking.  Sophie decieded to use rst due to problems with the md plugin for sphynx last year. It is somewhat annoying but I've gotten the hang of it fairly quickly.

I'll look into the conf.py differences but I'm assuming that's because we aren't using the md plugin and because it is just a template

**Comment by HagenSR** [comment]
You were right about the conf.py. I have also added the _static folder with override css. I can confirm that you can host this sphinx template on githuib

**Review by topoftheyear — APPROVED** [review]
It's good to see you know what you're doing. 

I can help you a bit with the conf.py. At the very bottom is the setup() method. You can see there's a 'url_resolver' in there. What this does is it allows for all the documentation to be visible entirely from the main website. Without it, clicking a link would take you straight to the github pages version.

File: sphinx/conf.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+# source_suffix = ['.rst', '.md']
+source_suffix = [ '.rst', '.md' ]
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = 'Byte-le-Royale 20XX'
+copyright = '20XX-20XX, NDSU ACM'
+author = 'NDSU ACM'
```
Comment (EricWM612) [inline sphinx/conf.py:53]: Don't forget this fun stuff 😄

File: sphinx/Makefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+github:
+	@make html
+	@cp -a _build/html/. ../docs
```
Comment (EricWM612) [inline sphinx/Makefile]: Is `-a` the flag you want to use here?

**Review by EricWM612 — APPROVED** [review]
Looks good to me, just dropped a couple of small comments in case they were missed. Not super familiar with `.rst` to make sphinx documentation, but I think it looks right. Unless you want me to be spellcheck too :P

File: sphinx/conf.py (hunk 1 above)
Comment (EricWM612) [reply sphinx/conf.py:53]: I somehow missed the fact that this was a template PR...ignore me

File: sphinx/Makefile (hunk 2 above)
Comment (HagenSR) [reply sphinx/Makefile]: Removed make file entirely.

**Pull request merged**